doc/services/zbus/index.rst

```diff
@@ -0,0 +1,356 @@
+
+Message delivery sequence
+-------------------------
+
+The listeners (synchronous observers) will follow the channel definition sequence as the notification and message consumption sequence. However, the subscribers, as they have an asynchronous nature, all will receive the notification as the channel definition sequence but only will consume the data when they assume the CPU again, so the delivery respects the order, but the reaction of it will be oriented by the priority assigned to the subscribers.
+
+Implementation
+**************
+
+Zbus operation depends on channels and observers. Therefore, it is necessary to determine its message and observers list during the channel definition. A message is a regular C struct; the observer can be a subscriber (asynchronous) or a listener (synchronous). Channels can have a combination of the following characteristics:
+
+* ``Validator function`` makes a channel accepts only valid messages;
+* ``Real-only flag`` makes a channel read-only.
+
+.. warning::
+
+    The ``Validator function`` will not take effect in read-only channels.
+
+The following code defines and initializes a read-only channel and its dependencies. This channel represents a firmware version, for example.
+
+.. code-block:: c
+
+    struct version {
+            uint8_t major;
+            uint8_t minor;
+            uint16_t build;
+    };
+
+    ZBUS_CHAN_DEFINE(version,	       /* Name */
+             true,		       /* Read only */
+             struct version,       /* Message type */
+
+             NULL,		       /* Validator */
+             ZBUS_OBSERVERS_EMPTY, /* observers */
+             ZBUS_MSG_INIT(.major = 0, .minor = 1,
+                       .build = 2) /* Initial value major 0, minor 1, build 1023 */
+    );
+
+
+The following code defines and initializes a regular channel and its dependencies. This channel exchanges accelerometer data, for example.
+
+.. code-block:: c
+
+    struct acc_msg {
+            int x;
+            int y;
+            int z;
+    };
+
+    ZBUS_CHAN_DEFINE(acc_data,				     /* Name */
+             false,					     /* Read only */
+             struct acc_msg,			     /* Message type */
+
+             NULL,					     /* Validator */
+             ZBUS_OBSERVERS(my_listener, my_subscriber), /* observers */
+             ZBUS_MSG_INIT(.x = 0, .y = 0, .z = 0)	     /* Initial value {0} */
+    );
+
+    void listener_callback_example(struct zbus_channel *chan)
+    {
+            struct acc_msg acc = {0};
+            if (&acc_data == chan) {
+                    zbus_chan_read(chan, &acc, K_NO_WAIT);
+                    LOG_DBG("From listener -> Acc x=%d, y=%d, z=%d", acc.x, acc.y, acc.z);
+            }
+    }
+
+    ZBUS_SUBSCRIBER_DEFINE(my_subscriber, 4);
+    void subscriber_task(void)
+    {
+            struct zbus_channel *chan = NULL;
+
+            while (!k_msgq_get(my_subscriber.queue, &chan, K_FOREVER)) {
```

Comment:
Have you considered adding a `zbus_...` function wrapping `k_msgq_get`? I'm going through the examples and it feels a bit out of place to use a msgq function to an internal field of `struct zbus_observer`, I'm thinking it may make sense to leave those fields opaque and have something that operates on the observer struct directly so you don't have to expose the `.queue` bit.